### PR TITLE
Feature/manual-scoring-init

### PR DESCRIPTION
**Pull request by iszmais**
Updated the manual scoring feature from  ILIAS 5.2 for up-to-date trunk (6.3.2019 16:33) with marked changes (Ticket #10580)

### CODE REVIEW

File: Modules/Test/classes/class.ilTestScoringByQuestionsGUI.php [hunk 1]
```diff
@@ -43,251 +43,282 @@ protected function getActiveSubTabId()
...
-						'participant'    => $participant,
-					);
+					if(false == isset($feedback['finalized_evaluation'])){
+						$feedback['finalized_evaluation'] = "";
+					}
+					
+					if(
+						isset($questionData['qid']) &&
+						$questionData['qid'] == $selected_questionData['question_id'] &&
+						($finalized_filter != 1 || $feedback['finalized_evaluation'] == 1) &&
```
Comment (legionth) [inline Modules/Test/classes/class.ilTestScoringByQuestionsGUI.php]: `$finalized_filter != 1 || $feedback['finalized_evaluation'] == 1` is very hard to understand for an outstanding developer. Maybe refactor this into a variable or method?

Same goes for the other statement.

File: Modules/Test/classes/class.ilTestScoringByQuestionsGUI.php [hunk 2]
```diff
@@ -43,251 +43,282 @@ protected function getActiveSubTabId()
...
+						($finalized_filter != 2 || $feedback['finalized_evaluation'] != 1)
```
Comment (legionth) [inline Modules/Test/classes/class.ilTestScoringByQuestionsGUI.php]: Where does these numbers come from? Does it make sense to put these into constants?

File: Modules/Test/classes/class.ilObjTest.php [hunk 3]
```diff
@@ -367,56 +367,56 @@ class ilObjTest extends ilObject implements ilMarkSchemaAware, ilEctsGradesEnabl
 *
 * @var int
 */
-	var $certificate_visibility;
+	public $certificate_visibility;
```
Comment (legionth) [inline Modules/Test/classes/class.ilObjTest.php]: Any reason why you change this?

File: Services/Form/classes/class.ilTextAreaInputGUI.php [hunk 4]
```diff
@@ -631,7 +631,104 @@ function insert($a_tpl)
 		$a_tpl->parseCurrentBlock();
 
 	}
-	
+
+	public function render()
```
Comment (mjansenDatabay) [inline Services/Form/classes/class.ilTextAreaInputGUI.php]: @iszmais Looks like there are some changes missing in `function insert($a_tpl)`.

File: Services/User/classes/class.ilObjUserFolder.php [hunk 5]
```diff
@@ -650,9 +650,11 @@ public static function _deleteAccountMailAttachment($a_lang)
 		include_once "Services/User/classes/class.ilFSStorageUserFolder.php";
 		$fs = new ilFSStorageUserFolder(USER_FOLDER_ID);
 		$path = $fs->getAbsolutePath()."/";
-		
-		unlink($path.$a_lang);
-		
+
```
Comment (mjansenDatabay) [inline Services/User/classes/class.ilObjUserFolder.php]: @iszmais / @legionth : Maybe the branch has to be rebased. This code hier does not belong to the desired code changes.

File: Services/Calendar/classes/class.ilCalendarSelectionBlockGUI.php [hunk 6]
```diff
@@ -409,6 +409,8 @@ protected function renderItem($a_set, $a_tpl)
 					break;
 			}
 
+			$ilCtrl->clearParameterByClass(ilCalendarPresentationGUI::class,'ref_id');
```
Comment (mjansenDatabay) [inline Services/Calendar/classes/class.ilCalendarSelectionBlockGUI.php]: @iszmais / @legionth : Maybe the branch has to be rebased. This code hier does not belong to the desired code changes.

File: Modules/TestQuestionPool/classes/class.assClozeTestGUI.php [hunk 7]
```diff
@@ -1253,6 +1253,7 @@ protected function getBestSolutionText($gap, $gap_index, $gap_combinations)
 
 	public function getAnswerFeedbackOutput($active_id, $pass)
 	{
+		$output = "";
```
Comment (mjansenDatabay) [inline Modules/TestQuestionPool/classes/class.assClozeTestGUI.php]: @iszmais Any reason for this change?

File: Modules/Test/classes/class.ilObjTest.php [hunk 8]
```diff
@@ -10546,79 +10546,199 @@ public function deliverPDFfromFO($fo, $title = null)
...
 	*
 	* @param integer $active_id Active ID of the user
 	* @param integer $question_id Question ID
 	* @param integer $pass Pass number
+	* @param boolean $force ignore access right
 	* @return string The feedback text
 	* @access public
 	*/
-	static function getManualFeedback($active_id, $question_id, $pass)
+	static function getManualFeedback($active_id, $question_id, $pass, $force = false)
```
[the remark this answers is not in the document]
Comment (iszmais) [reply Modules/Test/classes/class.ilObjTest.php]: Force to return the Manual Feedback without checking if its final.

[the remark this answers is not in the document]
Comment (legionth) [reply Modules/Test/classes/class.ilObjTest.php]: Then this should get a better name :)

**Pull request closed without merging**